workshops/discord_chatbot/README.md

```diff
@@ -0,0 +1,372 @@
+---
+name: Discord Chatbot
+description: Make a Discord bot with Discord.js and Dialogflow which talks with you!
+author: '@k4u5h4L'
+---
+
+# Discord Chatbot
+
+Make a discord bot which talks with you!
+
+| What you'll build       |
+| ----------------------- |
+| ![workshop demo](https://cloud-bu5ume7yu.vercel.app/0discord-chatbot-demo.gif) |
+
+References:
+
+- Here's the [final code](https://repl.it/@k4u5h4L/discord-chatbot-complete).
+
+- Here's the [starter code](https://repl.it/@k4u5h4L/discord-chatbot).
+
+The workshop should take about 20 minutes.
+
+The video tutorial for this workshop is linked at the bottom of this workshop with the required links for documentation and resources.
+
+I will leave all reference links below so that it will be easy for you to find it in one place
+
+- Bot: https://discord.js.org/
+- Authentication: https://cloud.google.com/docs/authentication/getting-started
+- Using Dialogflow: https://github.com/googleapis/nodejs-dialogflow#using-the-client-library
+- Dialogflow console: https://dialogflow.cloud.google.com/
+- Discord developers portal: https://discord.com/developers
+- Bot permissions calculator: https://discordapi.com/permissions.html
+
+## Part 1: Getting API keys!
+
+### Setting up the Discord bot
+
+To set up the bot, you need to visit the [Discord dev portal](https://discord.com/developers) and set up an application.
+Click to the top right button to create a new application.
+Keep this page open for a while, as we may be taking some stuff from here...
+
+![new app](https://cloud-9x2pb85rb.vercel.app/screenshot1.png)
+
+Go ahead and give it a cool name! I'm going to name it as doodle-noddle, but you can name it whatever you want!
+
+![fill name](https://cloud-9x2pb85rb.vercel.app/screenshot2.png)
+
+Then go ahead and create the application.
+Now go to the bot section in the menu on the left hand side, and add a new bot.
+
+![add bot](https://cloud-9x2pb85rb.vercel.app/screenshot3.png)
+
+If you get any formation like this, click on "yes, do it".
```

Comment:
"formation" is awkward wording, should be something like "popup" or "this message" instead